### DISCUSSION TITLE
Some Advanced Options Unexplained

### DISCUSSION THREAD

**Discussion by Chudroy** [nerdyrodent/VQGAN-CLIP#88] [Q&A]
Hi! There are some advanced parameters that have no explanation, is there some more in-depth guide to these parameters, or can anyone give a short explanation on use and parameter input?

  -in INIT_NOISE, --init_noise INIT_NOISE
 -iw INIT_WEIGHT, --init_weight INIT_WEIGHT
  -cuts CUTN, --num_cuts CUTN
  -cutp CUT_POW, --cut_power CUT_POW
 -o OUTPUT, --output OUTPUT

Can anyone give an explanation to these, and/or examples on how to use them? 

Thanks in advanced!

**Comment by tavisc** [comment]
I don't know about the first four but -o gives an output a specified name.

IE; -o test.jpg would create a jpg named test where ever you have your output being sent to.

**Comment by jonnm4** [comment]
INIT_NOISE sets the type of noise used when no initial_image is provided. It can be gradient or pixels by default it's on gradient.

CUTM can be "original", "updated", "nrupdated", "updatedpooling", "latest". none of which I can explain. By default it's "latest".

CUTN I'm not totally sure of yet though reducing it allows for larger images without memory issues. I don't see any impact on quality from the low cut number. as far as I can tell. By default it's 32.

CUT_POW should be the amount of influence cuts have on the picture; whatever cuts are. By default its 1.

Somebody elaborate I'd love to more about these cuts.

**Comment by jonnm4** [comment]
in Generate.py you'll find

perceptor = clip.load(args.clip_model, jit=jit)[0].eval().requires_grad_(False).to(device)
cut_size = perceptor.visual.input_resolution

which I like means; cuts are cutouts of the image the size of the image. later you find.

class MakeCutoutsUpdate(nn.Module):
    def __init__(self, cut_size, cutn, cut_pow=1.):
        super().__init__()
        self.cut_size = cut_size
        self.cutn = cutn
        self.cut_pow = cut_pow
        self.augs = nn.Sequential(
            K.RandomHorizontalFlip(p=0.5),
            K.ColorJitter(hue=0.01, saturation=0.01, p=0.7),
            # K.RandomSolarize(0.01, 0.01, p=0.7),
            K.RandomSharpness(0.3,p=0.4),
            K.RandomAffine(degrees=30, translate=0.1, p=0.8, padding_mode='border'),
            K.RandomPerspective(0.2,p=0.4),)
            
which make me think Cuts are copys of the image with random transforms applied and then blended back on.

**Comment by jonnm4** [comment]
![image](https://user-images.githubusercontent.com/20403377/191113611-d1fc56f0-757e-4a40-911d-3765a37d3c4f.png)
https://alexasteinbruck.medium.com/explaining-the-code-of-the-popular-text-to-image-algorithm-vqgan-clip-a0c48697a7ff
Image Cutouts are for CLIP but; why the random transform? Won't that just handicap CLIP from knowing what the image is.
